### ISSUE TITLE
"These set of vertices is not an identity group"

### ISSUE THREAD

**Issue by 19h** [fkie-cad/dewolf#386]
### What happened?

```
[Default] [pipeline.py:112 run()] ERROR - Failed to decompile geo::codec::decodeLocalizations, error during stage identity-elimination: At least two variables in the identity group {x0_11#11 (type: char * aliased: False), x20#43 (type: char * aliased: False), x20#44 (type: char * aliased: False), x21#6 (type: char * aliased: False), x20#2 (type: char * aliased: False), x20#6 (type: char * aliased: False), x21#3 (type: char * aliased: False), x0_6#6 (type: char * aliased: False), x20#1 (type: char * aliased: False), x20#5 (type: char * aliased: False)} have out degree zero, namely x0_6#6 and x20#1, i.e., these set of vertices is not an identity group
[Default] [identity_elimination.py:225 find_replacement_variable_of_group()] ERROR - At least two variables in the identity group {x0_11#11 (type: char * aliased: False), x20#43 (type: char * aliased: False), x20#44 (type: char * aliased: False), x21#6 (type: char * aliased: False), x20#2 (type: char * aliased: False), x20#6 (type: char * aliased: False), x21#3 (type: char * aliased: False), x0_6#6 (type: char * aliased: False), x20#1 (type: char * aliased: False), x20#5 (type: char * aliased: False)} have out degree zero, namely x0_6#6 and x20#1, i.e., these set of vertices is not an identity group
```

I don't have a copy of the decompilation output but it essentially said "decompilation failed".

### How to reproduce?

Load the file attached to https://github.com/fkie-cad/dewolf/issues/385#issue-2142653437 -- in this case I tried decompiling the function geo::codec::decodeLocalizations.

### Affected Binary Ninja Version(s)

4.0.4886

**Comment by ebehner** [comment]
related to #135

**Comment by ebehner** [comment]
Fixed on current main (commit 50d6b5298935d048be7656818226ea2850cd4614)
